hack/ci/Vagrantfile-flatcar

```diff
@@ -61,9 +61,11 @@ Vagrant.configure("2") do |config|
       crictl --runtime-endpoint="unix:///run/containerd/containerd.sock" images
 
       # Install Go
-      GO_VERSION=1.21
       mkdir /opt/go
-      curl -sSfL -o- https://go.dev/dl/go$GO_VERSION.0.linux-amd64.tar.gz | tar xfz - -C /opt
+      GO_VERSION=$(curl -sSfL "https://go.dev/VERSION?m=text" | head -n1)
+      curl -sSfL -o- https://go.dev/dl/$GO_VERSION.linux-amd64.tar.gz |
+        tar xfz - -C /opt
+      GO_VERSION=1.21
```

Comment:
Do we still need the second `GO_VERSION`?
```suggestion
```

Reply:
No, removed that.